semantic_version/base.py

```diff
@@ -1230,15 +1230,21 @@ class Parser:
 
         NUMBER = r'x|X|\*|0|[1-9][0-9]*'
         PART = r'[a-zA-Z0-9.-]*'
+        OP = r'<|<=|=|>=|>|\^|~|~>'
```

Comment:
IMHO the order matter ans the original code below may be buggy with `(?P<op><|<=|>=|>|=|\^|~|) `
This may be better, basically ordering longer operators first:
```suggestion
        OP = r'~>|<=|>=|>|<|\^|~|='
```

See https://docs.python.org/3/library/re.html?highlight=re#index-13

>  As the target string is scanned, REs separated by '|' are tried from left to right. When one pattern completely matches, that branch is accepted. This means that once A matches, B will not be tested further, even if it would produce a longer overall match. In other words, the '|' operator is never greedy.